### PR TITLE
fix shipping,domestic,internationl,total and escrow prices according …

### PR DESCRIPTION
**Pull request by vikas-cis**
…to country

#985

### CODE REVIEW

File: src/app/core/market/api/cart/cart.model.ts [hunk 1]
```diff
@@ -22,26 +22,34 @@ export class Cart {
       this.subTotal = new Amount(total);
     }
 
-    setShippingTotal(): void {
+    setShippingTotal(country: string): Amount {
       let total = 0.0;
       this.listings.forEach(listing => {
-        total += listing.internationalShippingPrice.getAmount();
+        total += country === listing.country ? (
```
Comment (kewde) [inline src/app/core/market/api/cart/cart.model.ts]: Wrong approach in this PR completely..

instead write a function the listing model:
listing.escrowPrice(country).getAmount()

Comment (vikas-cis) [reply src/app/core/market/api/cart/cart.model.ts]: @kewde what's wrong with the approach ? Before escrowPrice is set using `setEscrowPrice` inside the model and that function is also used `setTotal` due to that reason I used the same functions using those variables!!

**Comment by vikas-cis** [comment]
@kewde please have a look and let me know 

Thanks

File: src/app/core/market/api/template/template.model.ts [hunk 2]
```diff
@@ -142,4 +149,24 @@ export class Template {
     }
   }
 
+  shippingAmount(country: string): Amount {
```
Comment (kewde) [inline src/app/core/market/api/template/template.model.ts]: all this stuff belongs in the listing model.

It involves the country parameter, which only applies to listings..

File: src/app/core/market/api/template/template.model.ts [hunk 3]
```diff
@@ -142,4 +149,24 @@ export class Template {
...
+    return this.object.country === country ? (
+          this.domesticShippingPrice
+        ) : (
+          this.internationalShippingPrice
+        )
+  }
+
+  escrowAmount(country: string): Amount {
+    return this.object.country === country ? this.escrowPriceDomestic : this.escrowPriceInternational;
```
Comment (kewde) [inline src/app/core/market/api/template/template.model.ts]: write a function `isDomestric(country)` to check equality

File: src/app/market/buy/checkout-process/checkout-process.component.html [hunk 4]
```diff
@@ -252,11 +252,11 @@
             {{listing.basePrice?.getIntegerPart()}}{{listing.basePrice?.dot()}}<small>{{listing.basePrice?.getFractionalPart()}}</small>
             PART
           </td>
-          <td class="shipping cell">{{listing.internationalShippingPrice?.getIntegerPart()}}{{listing.internationalShippingPrice?.dot()}}<small>{{listing.internationalShippingPrice?.getFractionalPart()}}</small>
+          <td class="shipping cell">{{listing?.shippingAmount(valueOf('country')).getIntegerPart()}}{{listing?.shippingAmount(valueOf('country')).dot()}}<small>{{listing?.shippingAmount(valueOf('country')).getFractionalPart()}}</small>
```
Comment (kewde) [inline src/app/market/buy/checkout-process/checkout-process.component.html]: valueOf() has been removed.

Use this instead `shippingFormGroup.values.country || ''`

File: src/app/core/market/api/template/template.model.ts (hunk 2 above)
Comment (vikas-cis) [reply src/app/core/market/api/template/template.model.ts]: ohhk

File: src/app/market/buy/checkout-process/checkout-process.component.html [hunk 5]
```diff
@@ -251,11 +251,11 @@
             {{listing.basePrice?.getIntegerPart()}}{{listing.basePrice?.dot()}}<small>{{listing.basePrice?.getFractionalPart()}}</small>
             PART
           </td>
-          <td class="shipping cell">{{listing.internationalShippingPrice?.getIntegerPart()}}{{listing.internationalShippingPrice?.dot()}}<small>{{listing.internationalShippingPrice?.getFractionalPart()}}</small>
+          <td class="shipping cell">{{listing?.shippingAmount(shippingFormGroup.value.country || '')?.getIntegerPart()}}{{listing?.shippingAmount(shippingFormGroup.value.country || '')?.dot()}}<small>{{listing?.shippingAmount(shippingFormGroup.value.country || '')?.getFractionalPart()}}</small>
```
Comment (kewde) [inline src/app/market/buy/checkout-process/checkout-process.component.html]: What about 
public country = this.shippingFormGroup.value.country || '';

and then just referencing the country variable, short and sweet.

File: src/app/market/listings/preview-listing/preview-listing.component.html [hunk 6]
```diff
@@ -41,10 +41,11 @@
           Local in {{ data?.listing?.country }} &ndash; {{ data?.listing?.domesticShippingPrice.getAmount() }} PART
         </td>
       </tr>
-      <tr>
+      <!-- @TODO how we calculate escrow as we don't know the shipping country -->
```
Comment (kewde) [inline src/app/market/listings/preview-listing/preview-listing.component.html]: That's a very good question. 
How do know which country price to pick IF we only get the addresses in the second step of the checkout.
Bring it up with the UX guys, @AllienWorks

Comment (HenkSwardt) [reply src/app/market/listings/preview-listing/preview-listing.component.html]: We had a discussion around this question on the group a couple days back. @xludx mentioned if there has been a previous order, we might be able to get this to work.

File: src/app/core/market/api/cart/cart.model.ts [hunk 7]
```diff
@@ -18,45 +18,42 @@ export class Cart {
...
-        total += listing.internationalShippingPrice.getAmount();
-      });
-      this.shippingTotal = new Amount(total);
-    }
+  setShippingTotal(country: string): Amount {
+    let total = 0.0;
+    this.listings.forEach(listing => {
+      total += listing.shippingAmount(country).getAmount();
+    });
+    return new Amount(total);
```
Comment (kewde) [inline src/app/core/market/api/cart/cart.model.ts]: Don't just do the small refactor on setSubTotal but everwhere lol

File: src/app/core/market/api/cart/cart.model.ts [hunk 8]
```diff
@@ -18,45 +18,42 @@ export class Cart {
...
+  }
 
-    setEscrowTotal(): void {
-      let total = 0.0;
-      this.listings.forEach(listing => {
-        total += listing.escrowPrice.getAmount();
-      });
-      this.escrowTotal = new Amount(total);
-    }
+  setEscrowTotal(country: string): Amount {
```
Comment (kewde) [inline src/app/core/market/api/cart/cart.model.ts]: get

File: src/app/core/market/api/cart/cart.model.ts [hunk 9]
```diff
@@ -18,45 +18,42 @@ export class Cart {
...
+  }
 
-    setShippingTotal(): void {
-      let total = 0.0;
-      this.listings.forEach(listing => {
-        total += listing.internationalShippingPrice.getAmount();
-      });
-      this.shippingTotal = new Amount(total);
-    }
+  setShippingTotal(country: string): Amount {
```
Comment (kewde) [inline src/app/core/market/api/cart/cart.model.ts]: get

File: src/app/core/market/api/cart/cart.model.ts [hunk 10]
```diff
@@ -18,45 +18,42 @@ export class Cart {
...
+    });
+    return new Amount(total);
+  }
 
-    setTotal(): void {
-      this.total = new Amount(this.subTotal.getAmount()
-               + this.shippingTotal.getAmount()
-               + this.escrowTotal.getAmount())
-    }
+  setTotal(country: string): Amount {
```
Comment (kewde) [inline src/app/core/market/api/cart/cart.model.ts]: get

**Pull request merged**